services/src/handlers/datasets.rs

```diff
@@ -97,8 +139,189 @@ fn adjust_user_path_to_upload_path(meta: &mut MetaDataDefinition, upload: &Uploa
     Ok(())
 }
 
+pub(crate) fn auto_create_dataset_handler<C: Context>(
+    ctx: C,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::path!("dataset" / "auto")
+        .and(warp::post())
+        .and(authenticate(ctx.clone()))
+        .and(warp::any().map(move || ctx.clone()))
+        .and(warp::body::json())
+        .and_then(auto_create_dataset)
+}
+
+// TODO: move into handler once async closures are available?
+async fn auto_create_dataset<C: Context>(
+    session: Session,
+    ctx: C,
+    create: AutoCreateDataSet,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    let upload = ctx
+        .data_set_db_ref()
+        .await
+        .get_upload(session.user.id, create.upload)
+        .await?;
+
+    let create = create.validated()?.user_input;
+
+    let main_file_path = upload.id.root_path()?.join(&create.main_file);
+    let meta_data = auto_detect_dataset(&main_file_path)?;
+
+    let properties = AddDataSet {
+        id: None,
+        name: create.dataset_name,
+        description: create.dataset_description,
+        source_operator: "OgrSource".to_owned(),
```

Comment:
I guess this should also be output of `auto_detect_dataset`.

Reply:
I thought about it but I think it makes more sense to output only the DataSetId, because we not always need the dataset after creation and then we would needlessly send out this information. It's also more in line with how the other API calls work

Reply:
But now, you just assume that after `auto_detect_dataset` it is always OgrSource.

Reply:
73b4e8430add8c17b2e5670d7a011c6fbc3fa5e4